tools/codegen/README.md

```diff
@@ -78,3 +74,100 @@ update-codegen-crds:
 
 You may also want to add the `_output` directory to your `.gitignore` to avoid checking in compiled binaries created
 by this make target.
+
+## Generators
+
+The following section describes the individual generators included within the `codegen` utility.
+
+The generators enabled by default are:
+- [Compatibility](#compatibility)
+- [Schemapatch](#schemapatch)
+- [Swaggerdocs](#swaggerdocs)
+
+### Compatibility
+
+To generate API compatibility comments, use the `compatibility` subcommand.
+
+The `compatibility` subcommand generates a compatibility level comment for each API defintion.
+The generation is controlled by a marker applied to the CRD struct defintiion.
+For example, this annotation would be `+openshift:compatibility-gen:level=1` for a level 1 API.
+	
+Valid API levels are 1, 2, 3 and 4. Version 1 is required for all stable APIs.
+Version 2 should be used for beta level APIs. Levels 3 and 4 may be used for alpha APIs.
```

Comment:
note for future us.  openshift/api should never again have a beta api version.  We're either devpreview and alpha, techpreview and alpha, or supported and stable.

Reply:
Yes I had noticed this as a tendency, I intend to add a section to the API conventions about when to use alpha vs stable APIs and will make a note in that about not using beta APIs